matrix.py

```diff
@@ -0,0 +1,156 @@
+                                 for column_i in range(len(other.data))]
+                return Matrix(tuple(splitting(post_add_list, len(self.data))))
+
+            else:
+                print("matrices must be equal length")
+
+        else:
+            print("cannot add a scalar")
+
+    def __sub__(self, other):
+        """Subtracts 2 matrices."""
+
+        if isinstance(other, Matrix):
+            if len(self.data) == len(other.data):
+                post_sub_list = [self.data[row_i][column_i]
+                                 - other.data[row_i][column_i]
+                                 for row_i in range(len(self.data))
+                                 for column_i in range(len(other.data))]
+                return Matrix(tuple(splitting(post_sub_list, len(self.data))))
+
+            else:
+                print("matrices must be equal length")
+
+        else:
+            print("cannot subtract from a / a scalar")
+
+    def __mul__(self, other):
+        """Multiplies matrix by scalar/matrix."""
+
+        if isinstance(other, Matrix):
+            if len(self.data) == len(other.data):
+                # Converting the second matrix to columns
+                column_matrix = [tup[index] for index in range(len(self.data))
+                                 for tup in other.data]
+                split_column_matrix = splitting(column_matrix, len(self.data))
+                post_mul = []
+                for tup in self.data:
+                    for split_list in split_column_matrix:
+                        for num, index in \
+                                zip(split_list, range(len(self.data))):
+                            post_mul.append(num * tup[index])
+                split_mul = splitting(post_mul,
+                                      len(self.data) * len(self.data))
+                added = [sum(tuples) for tuples in split_mul]
+                final_mtx = splitting(added, len(self.data))
+                return Matrix(tuple(final_mtx))
+            else:
+                print("cannot multiply different dimension matrices")
+
+        else:
+            final_mtx = [num * other for tup in self.data for num in tup]
+            return Matrix(tuple(splitting(final_mtx, len(self.data))))
+
+    def __truediv__(self, other):
+        """Divide matrix by scalar."""
+
+        if not isinstance(other, Matrix):
+            divided_mtx = [num / other for tup in self.data for num in tup]
+            return Matrix(tuple(splitting(divided_mtx, len(self.data))))
+
+        else:
+            print("cannot divide matrix by matrix")
+
+    @staticmethod
+    def unity(num):
+        """Display tuple of tuples with 1 & 0
+        so that the ones create a diagonal if seen as a square.
+
+            Args:
+                num: the amount of ones the user wants
+
+            Returns: the final tuple, as a Matrix type
+        """
+
+        unity_list = [0 for _ in range(num * num)]
```

Comment:
why not
```suggestion
        unity_list = [0] * (num*num)
```

**do the same to line 128**